src/configuration/index.ts

```diff
@@ -128,27 +130,57 @@ export class Configuration implements IConfiguration {
 			dotAndEnv = loadash.merge({}, envConfig, dotAndEnv);
 		}
 
-		// read configuration files, first default.json, then NODE_ENV.json (which defaults to development.json)
-		const configurationFileNames = [];
-		const configurations = [];
-		if ('NODE_ENV' in dotAndEnv) {
-			this.NODE_ENV = dotAndEnv['NODE_ENV'];
+		// set default NODE_ENV
+		if (NODE_ENV in dotAndEnv) {
+			this.NODE_ENV = dotAndEnv[NODE_ENV];
 		} else {
 			this.NODE_ENV = this.options.defaultNodeEnv;
 		}
+
+		// read configuration files, first default.json, then NODE_ENV.json (which defaults to development.json)
+		const configurationFileNames: string[] = [];
+		const configurationFileNamesAddedSuccessfully = [];
+		const configurations = [];
+		// start with default file
 		configurationFileNames.push('default.json');
-		if (this.NODE_ENV !== 'default') {
-			configurationFileNames.push(this.NODE_ENV + '.json');
+		// add environment files
+		if (
+			Array.isArray(this.options.loadFilesFromEnv) &&
+			this.options.loadFilesFromEnv.length != 0
+		) {
+			this.options.loadFilesFromEnv.forEach((envName) => {
+				let ok = true;
+				// add configuration based on current envName, ignore default already added first
+				if (!(envName in dotAndEnv)) {
+					ok = false;
+					logger.warn("ignore envName, it's not defined", envName);
+				}
+				if (configurationFileNames.includes(dotAndEnv[envName])) {
+					ok = false;
+					logger.warn('ignore envName, already added value before');
+				}
+				if (ok) {
+					configurationFileNames.push(dotAndEnv[envName]);
+				}
+			});
 		}
+		logger.info(
+			'will parse following configuration filenames in given order',
+			configurationFileNames
+		);
 		for (const file of configurationFileNames) {
 			const fullFileName = path.join(
 				this.options.baseDir,
 				this.options.configDir,
-				file
+				file + '.json'
```

Comment:
file,
'.json'

Reply:
I am not sure what it is loaded in default case or with some settings. We should speak about it later the day.

Reply:
Currently, JSON files are supported only. The extension was added before into configurationFileNames for every single entry which has been changed here.

Reply:
adding json as parameter would finally handle file as folder name.